src/jit/compiler.hpp

```diff
@@ -1456,20 +1456,38 @@ inline void GenTree::SetOper(genTreeOps oper, ValueNumberUpdate vnUpdate)
     }
 }
 
-inline void GenTree::CopyFrom(const GenTree* src, Compiler* comp)
+//------------------------------------------------------------------------
+// CopyFrom: replace this with the src node. If the source is isolated node, then use it as it.
```

Comment:
I thought about that, in that case I will rename function to `ReplaceWith`.

Reply:
When our nodes are not linked these gtPrev and gtNext checks are not valuable. Do we have a way to express that src node is in incorrect state and should not be used? Like set gtTreeId to -1 or smth like that.

Reply:
> Do we have a way to express that src node is in incorrect state and should not be used? Like set gtTreeId to -1 or smth like that.

I believe that setting the `gtOper` to `GT_NONE` might be a reasonable approach.

Reply:
There is also DEBUG_DESTROY_NODE which is the recommended way to mark a node that should no longer be referenced in the graph of nodes.

Reply:
FYI:

/*****************************************************************************
 * This node should not be referenced by anyone now. Set its values to garbage
 * to catch extra references
 */

inline void DEBUG_DESTROY_NODE(GenTreePtr tree)
{
#ifdef DEBUG
    // printf("DEBUG_DESTROY_NODE for [0x%08x]\n", tree);

    // Save gtOper in case we want to find out what this node was
    tree->gtOperSave = tree->gtOper;

    tree->gtType = TYP_UNDEF;
    tree->gtFlags |= 0xFFFFFFFF & ~GTF_NODE_MASK;
    if (tree->OperIsSimple())
    {
        tree->gtOp.gtOp1 = tree->gtOp.gtOp2 = nullptr;
    }
    // Must do this last, because the "gtOp" check above will fail otherwise.
    // Don't call SetOper, because GT_COUNT is not a valid value
    tree->gtOper = GT_COUNT;
#endif
}

Reply:
Thanks, it is very helpful.

Reply:
> There is also DEBUG_DESTROY_NODE which is the recommended way to mark a node that should no longer be referenced in the graph of nodes.

Thanks, Brian - I'd forgotten about that!